spec/presenters/mainstream_browse_page_presenter_spec.rb

```diff
@@ -102,5 +108,52 @@
         end
       end
     end
+
+    describe "linking to other pages" do
+
+      # top level pages
+      let(:alpha) { create(:mainstream_browse_page, title: "alpha") }
+      let(:bravo) { create(:mainstream_browse_page, title: "beta") }
+
+      # second level pages
+      let(:delta) { create(:mainstream_browse_page, title: "delta") }
+      let(:echo) { create(:mainstream_browse_page, title: "echo") }
+      let(:foxtrot) { create(:mainstream_browse_page, title: "foxtrot") }
+
+      # topics
+      let(:golf) { create(:topic, :title => "Golf") }
+      let(:hotel) { create(:topic, :title => "Hotel") }
+
+
+      let(:presenter) { MainstreamBrowsePagePresenter.new(delta) }
+      let(:presented_data) { presenter.render_for_publishing_api }
+
+      before :each do
+        # for being considered a top level page, the page must have children
+        alpha.children = [delta, echo]
+        bravo.children = [foxtrot]
+
+        delta.topics = [golf, hotel]
+
+        alpha.save!
+        delta.save!
+        echo.save!
+      end
+
+      it "Delta has all the possible linked pages" do
```

Comment:
I think these tests could be structured more clearly. It's not clear what 'Delta' is here.  It would also be good to cover this for both a top-level browse page, and a second-level page.  Maybe something like:

```ruby
context "for a top-level browse page" do
  it "includes all the top-level browse pages"
  it "includes all the child pages for the top-level page"
  it "is valid against the schema"
end

context "for a second-level browse page" do
  it "includes all the top-level browse pages"
  it "includes the parent top-level browse page"
  it "includes all the second-level pages that are children of the parent page"
  it "is valid against the schema"
end
```